src/devices/somfy_iohc.c

```diff
@@ -91,60 +91,123 @@ Example packets:
 
 #include "decoder.h"
 
+struct iohc_msg {
+    /* Mandatory fields */
+    /* Control byte 1 */
+    unsigned end_flag : 1;
+    unsigned start_flag : 1;
+    unsigned protocol_mode : 1;
+    unsigned frame_length : 5;
+    /* Control byte 2 */
+    unsigned use_beacon : 1;
+    unsigned is_routed : 1;
+    unsigned low_power_mode : 1;
+    unsigned protocol_version : 3;
+    /* Addresses */
+    unsigned dst_addr : 24;
+    unsigned src_addr : 24;
+    /* Command ID */
+    unsigned cmd_id : 8;
+    char data[31 * 2 + 1]; /* variable length, converted to hex string */
+    unsigned crc : 16;
+
+    /* optional fields */
+    int seq_nr;
+    char mac[13];
+};
+
 static int somfy_iohc_decode(r_device *decoder, bitbuffer_t *bitbuffer)
 {
     uint8_t const preamble_pattern[] = {0x57, 0xfd, 0x99};
+    struct iohc_msg msg = {0};
 
-    uint8_t b[19 + 15]; // 19 byte + up 15 byte payload
+    uint8_t b[1 + 31 + 2]; // Length, payload, CRC
 
     if (bitbuffer->num_rows != 1)
         return DECODE_ABORT_EARLY;
 
-    unsigned offset = bitbuffer_search(bitbuffer, 0, 0, preamble_pattern, 24) + 24;
-    if (offset + 19 * 10 >= bitbuffer->bits_per_row[0])
+    unsigned offset = bitbuffer_search(bitbuffer, 0, 0, preamble_pattern, 24);
+    if (offset == bitbuffer->bits_per_row[0])
+        return DECODE_ABORT_EARLY;
+
+    offset += 24;
+
+    int num_bits = bitbuffer->bits_per_row[0] - offset;
+    if (num_bits <= 0)
         return DECODE_ABORT_EARLY;
 
-    unsigned num_bits = bitbuffer->bits_per_row[0] - offset;
-    num_bits = MIN(num_bits, sizeof (b) * 8);
+    if ((size_t)num_bits > sizeof b * 8)
```

Comment:
keep the parens on sizeof and add braces for the block, otherwise maybe keep `MIN`

Reply:
Why braces? Other single-line blocks don't have any. `sizeof` is an operator, but well... I'll change it back to MIN with a cast.

Reply:
Operator, yes, that's why `sizeof (x)` is prefered to `sizeof(x)`. The `sizeof x * y` just looks sketchy at a quick glance.
Newer and changed code should preferably use braces, even for single-line blocks -- the compiler and linter will warn on mishaps, still  it's just an accident waiting to happen.